tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py

```diff
@@ -167,6 +168,82 @@ async def test_should_not_revert_when_address_is_not_zero(
                 caller_address=addresses[0].starknet_address,
             )
 
+    class TestExceptionHandling:
+        async def test_revert_should_always_revert(self, plain_opcodes, owner):
+            with kakarot_error("This always reverts"):
+                await plain_opcodes.testShouldAlwaysRevert(
+                    caller_address=owner.starknet_address,
+                )
+
+        async def test_revert_should_revert_contract_writes(
+            self, counter, caller, owner
+        ):
+            success, data = await caller.call(
+                target=counter.evm_contract_address,
+                payload=counter.encodeABI("doRevert"),
+                caller_address=owner.starknet_address,
+            )
+
+            # we ignore the first 4 bytes because it is the function selector
+            # see conventions for the structure of return payloads on reverting cases
+            # https://docs.soliditylang.org/en/latest/control-structures.html#revert
+            assert "Revert reason" == Web3().codec.decode(["string"], data[4:])[0]
+            assert success == False
+            assert await counter.count() == 0
+            # TODO: is this just asserting that there were no events firing from a contract,
+            # or that there were no events fired from all touched contracts in an execution?
+            assert caller.events._events == []
```

Comment:
actually this `_events` key is from Web3 but not updated by the kakarot wrapper. The wrap_kakarot only `setattr` matching emitted events of the given contract
see https://github.com/sayajin-labs/kakarot/blob/91999728588bfa9efb047b16c2972dfa4e83838b/tests/integration/helpers/wrap_kakarot.py#L95

in other words, currentl, events emitted in subcontexts would be filtered out in the wrapper so there is no way to assert here something about them.

Because it's difficult from the wrapper to know which sub context could have emitted, I think that an option would to add the raw `log_receipts` from the tx and a method to filter these receipts for any given target even abi